apps/angular/src/core/services/anime.service.ts

```diff
@@ -17,14 +22,29 @@ import { ApiUrlService } from './api-url.service';
 export class AnimeService {
 	private readonly http = inject(HttpClient);
 
-	private readonly urlService = inject(ApiUrlService);
+	private readonly apiUrlService = inject(ApiUrlService);
 
-	/**
-	 * Get a list of anime from the API.
-	 */
-	public getAnimeList(): Observable<Anime[]> {
-		return this.http.get<PaginationDto<AnimeDto>>(this.urlService.animeListPath).pipe(
-			map(animeDto => animeDto.results.map(anime => AnimeMapper.fromDto(anime))),
+	private readonly pageParamsService = inject(PageParamsService);
+
+	private readonly destroyRef = inject(DestroyRef);
+
+	private readonly animeParams$ = this.pageParamsService.animeParams$;
+
+	/** Get a list of anime from the API. */
+	public getAnimeList(): Observable<Pagination<Anime>> {
+		return this.animeParams$.pipe(
+			takeUntilDestroyed(this.destroyRef),
+			switchMap(params => {
+				const httpParams = AnimeParamsMapper.toAnimeHttp(params);
+
+				return this.http.get<PaginationDto<AnimeDto>>(this.apiUrlService.animeListPath, { params: httpParams }).pipe(
+					map(pagination => PaginationMapper.fromDto(
+						pagination,
+						animeDto => AnimeMapper.fromDto(animeDto),
+					)),
+				);
+			}),
```

Comment:
same

Reply:
Let's make the data transfer from the component to the service component explicit.

Reply:
What do you think of this approach? Now we are more explicit that we are using params as an observable and not changing params.
```suggestion
private readonly animeParams$ = this.pageParamsService.getAnimeParams();
```

Reply:
You do not need to store filter parameters in the service at all. Just pass them as a method argument